### DISCUSSION TITLE
Allow magic search on local files that are not backuped up to ente

### DISCUSSION THREAD

**Discussion by rshxyz** [ente-io/ente#1356] [Enhancements]
I would like to run magic search on local files on the device that I have not backed up on ente

**Comment by vishnukvmd** [comment]
Hey, thanks for the feedback. The way the system is designed right now the embeddings for magic search are tied to uploaded files, so it will take a bit of work to work around this limitation. But I very much get the use case, will keep this discussion open so that it can gain traction.

**Comment by lakshay487** [comment]
One year later, it's still not implemented.

Here are some models that can be used for it if the already existing model can't: 
- open clip with the Laion2B dataset https://github.com/mlfoundations/open_clip
- long clip (bigger token lengths during training) https://github.com/beichenzbc/Long-CLIP
- meta-clip https://github.com/facebookresearch/metaclip
- google siglip https://huggingface.co/docs/transformers/main/en/model_doc/siglip different loss function
- apple dfn https://huggingface.co/apple/DFN5B-CLIP-ViT-H-14 if I recall the 5b uses a higher image resolution, amongst a number of other innovations (378 ×378 resolution)
- Blip 1 & 2 by salesforce https://huggingface.co/docs/transformers/en/model_doc/blip
